File: Problems/Week_3/4.8_단어_공부(1157)/최호/1157.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class WordStudy {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in)) ;
        String Word = br.readLine().toUpperCase();
        int max = 0;
        int index;
        int[] count = new int[26];
        char result = ' ';

        for(int i = 0; i < Word.length(); i++){
            index = Word.charAt(i)-65;
            count[index]++;

            if(max < count[index]){
                max = count[index];
                result = Word.charAt(i);
            }
            else if(max == count[index]){
                result = '?';
            }
        }
        System.out.println(result);

    }
}

File: Problems/Week_3/4.8_단어_공부(1157)/최진영/1157.c
#define _CRT_SECURE_NO_WARNINGS
#include <stdio.h>
int main(void)
{
	char word[1000000] = {0,};
	int count[91] = { 0, };
	int result;
	scanf("%s", word);
	for (int w = 0;; w++)
	{
		if (word[w] > 96)
		{
			word[w]= word[w]-32;
		}
		else if (word[w] == 0)
		{
			break;
		}
	}

	for (int v=0;;v++)
	{
		if (word[v] == 0)
		{
			break;
		}
			count[(int)word[v]] += 1;
	}

	for (int t = 65; t < 91; t++)
	{
		if (count[t] > count[t - 1])
		{
			result = t;
		}
		else if (count[t] < count[t - 1])
		{
			count[t] = count[t - 1];
		}
		else if (count[t] || count[t - 1])
		{
			result = 63;
		}
	}
	printf("%c", (char)result);
	return 0;
}